### ISSUE TITLE
Rule Purposal: Prolog Directives

### ISSUE THREAD

**Issue by ChristianMurphy** [eslint/eslint#5657]
**When the rules will warn. Include a description as well as sample code.**

**NOTE:** This is likely a wishlist item for ESLint 3

This rule would be a more generic version of the existing `strict` rule, allowing for directives other than `'use strict'` and for multiple directives to be set.

Rule would allow for a list of directives to be given, as well as what scope would allow the directive

***valid***
``` javascript
/* eslint prolog-directives: [2, [{"directive":"use strict", "scope": "global"}]] */

'use strict';

console.log('example');
```

``` javascript
/* eslint prolog-directives: [2, [{"directive":"use strict", "scope": "global"}, {"directive":"use strong", "scope": "function"}]] */

'use strict';

function example() {
   'use strong';
   console.log('example');
}
```

***invalid***
``` javascript
/* eslint prolog-directives: [2, [{"directive":"use strict", "scope": "global"}]] */

console.log('example');
```

``` javascript
/* eslint prolog-directives: [2, [{"directive":"use asm", "scope": "never"}]] */

'use asm';

console.log('example');
```

**Whether the rule prevents an error or is stylistic?**
This would prevent errors.

**Why the rule should be in the core instead of creating a custom rule?**
This rule could replace the existing `strict` rule, with a more generic rule, that would allow for any directive without creating new rules.

**Are you willing to create the rule yourself?**
Yes

**Additional Resources**
Supercedes: #5655
Related #2240
ASM JS: http://asmjs.org/spec/latest/#validatemodule-f
Strong Mode: https://github.com/v8/v8/wiki/Experiments%20with%20Strengthening%20JavaScript#strong-mode
V8/Node supported directives: https://github.com/nodejs/node/blob/v4.x/deps/v8/src/ast-value-factory.h#L283-L285

**Comment by nzakas** [comment]
Since there's currently only one formal JavaScript directive, "use strict," this is not a good candidate to be included as a core rule. I'd suggest creating a custom rule if you need this functionality.
